src/AVFoundation/AVAudioPlayer.cs

```diff
@@ -23,67 +23,57 @@
 //
 using Foundation;
 using ObjCRuntime;
-using AudioToolbox;
 using System;
+using System.Runtime.InteropServices;
 
 #nullable enable
 
 namespace AVFoundation {
 #if !WATCH
 	public partial class AVAudioPlayer {
 
+		[DllImport (Constants.ObjectiveCLibrary, EntryPoint = "objc_msgSend")]
+		unsafe static extern IntPtr objc_msgSend (IntPtr receiver, IntPtr selector, IntPtr arg1, IntPtr arg2);
+
 		public static AVAudioPlayer? FromUrl (NSUrl url, out NSError? error)
 		{
+			error = null;
+			IntPtr url__handle__ = url!.GetNonNullHandle (nameof (url));
+			IntPtr handle = Messaging.IntPtr_objc_msgSend (class_ptr, Selector.GetHandle ("alloc"));
+			IntPtr errorptr = IntPtr.Zero;
+			if (handle == IntPtr.Zero)
+				return null;
 			unsafe {
-				IntPtr errhandle;
-				IntPtr ptrtohandle = (IntPtr) (&errhandle);
-
-				var ap = new AVAudioPlayer (url, ptrtohandle);
-				if (ap.Handle == IntPtr.Zero) {
-					error = Runtime.GetNSObject<NSError> (errhandle);
-					return null;
-				} else
-					error = null;
-				return ap;
+				handle = objc_msgSend (handle, Selector.GetHandle ("initWithContentsOfURL:error:"), url__handle__, errorptr);
```

Comment:
And then the last argument here you pass the address of `errorptr`:

```suggestion
				handle = objc_msgSend (handle, Selector.GetHandle ("initWithContentsOfURL:error:"), url__handle__, &errorptr);
```